e2e_test/sink/sink_into_table/alter_column.slt

```diff
@@ -178,6 +178,9 @@ select * from t_append_only;
 statement ok
 create sink s21 into t_append_only as select v1+1000 as v1, v1+2000 as v2, v1+3000 as v3 from t_s2 with (type = 'append-only', force_append_only = 'true');
 
+statement ok
+recover;
+
```

Comment:
Why do we add some `recover` here? While this covers the path for re-constructing the sink-into-table plan, will it also make some existing path inaccessible?